modules/apps/frontend-js/frontend-js-components-sample-web/src/main/resources/META-INF/resources/js/TranslationManagerSampleReact.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLayout from '@clayui/layout';
+import {TranslationAdminSelector} from 'frontend-js-components-web';
+import React from 'react';
+
+export default function TranslationManagerSampleReact({
+	activeLanguageIds,
+	availableLocalesJsp,
+	defaultLanguageId,
+	translations,
+}) {
+	const availableLocales = availableLocalesJsp.map((item) => {
```

Comment:
Since this is a demo module and we are using this react component/file just once it should be the same as having it outside in the jsp or java file, right?

Reply:
In terms of final results, yes, is the same. In terms of performance is not. 

We need to ask ourselves if the effort here is trivial enough compared with the benefits (like performance, readability and manteinance) and the future escalation.

What do you think?

Reply:
I can see why the maintenance will improve but I can't say the same about the performance. To me it's the same to "construct" the object in the JS file as it is in the JSP, since it will be called once in this module.

Having said that, I'm going to move this part to the JSP.